src/runner/trainers/acdc_seg_trainer.py

```diff
@@ -0,0 +1,51 @@
+import torch.nn.functional as F
+
+from src.runner.trainers import BaseTrainer
+
+
+class AcdcSegTrainer(BaseTrainer):
+    """The ACDC trainer for the segmentation task.
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def _train_step(self, batch):
+        input, target = batch['input'].to(self.device), batch['target'].to(self.device)
+        output = self.net(input)
+        cross_entropy_loss = self.loss_fns.cross_entropy_loss(output, target.squeeze(dim=1))
+        dice_loss = self.loss_fns.dice_loss(output, target)
+        loss = (self.loss_weights.cross_entropy_loss * cross_entropy_loss
+                + self.loss_weights.dice_loss * dice_loss)
+        return {
+            'loss': loss,
+            'losses': {
+                'CrossEntropyLoss': cross_entropy_loss,
+                'DiceLoss': dice_loss
+            }
+        }
+
+    def _valid_step(self, batch):
+        input, target = batch['input'].to(self.device), batch['target'].to(self.device)
+        output = F.interpolate(self.net(input),
```

Comment:
I'm not sure what is "dynamic upsampling" you mentioned. Besides, I haven't tried other solution because the results are reasonable.

Reply:
`dynamic upsampling` means we dynamically adjust the shape of feature maps before they are concatenated by the skip connection.

Reply:
Sorry I'm quite confused. Any reference for this?

Reply:
![image](https://user-images.githubusercontent.com/11289377/78469601-5f1ec900-7755-11ea-8bbe-5062249d995a.png)

If using the `_UpBlock` in the network, why do we need to interpolate the output of the network into the shape of the target? The shapes of the output and the target should be the same, right?

Reply:
Actually it's not because the input data are isotropic (1mm x 1mm x 1mm) while the target is the original resolution. That's why we need to interpolate the network output.

Reply:
Got it. I misunderstand.